### PR TITLE
chore: release v0.18.0

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

**Review by lexnv — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,79 @@ The format is based on [Keep a Changelog].
...
+
+
+#[async_trait]
+impl RpcServer for () {
+	async fn keys(&self) -> Result<Vec<StorageKey>, Error> {
+		Error::A
+	}
+}
+```
+
```
[the remark this answers is not in the document]
Comment (niklasad1) [reply CHANGELOG.md:78]: oops forgot that :D

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,79 @@ The format is based on [Keep a Changelog].
...
+enum Error {
+	A,
+	B,
+}
+
+#[async_trait]
+impl RpcServer for () {
+	async fn keys(&self) -> Result<String, jsonrpsee::core::Error> {
+		Err(jsonrpsee::core::Error::to_call_error(Error::A))
+		// or jsonrpsee:::core::Error::Call(CallError::Custom(ErrorObject::owned(1, "a", None::<()>)))
```
Comment (niklasad1) [inline CHANGELOG.md:41]:
```suggestion
		// or jsonrpsee::core::Error::Call(CallError::Custom(ErrorObject::owned(1, "a", None::<()>)))
```

**Review by jsdw — APPROVED** [review]
Love the summary and good stuff :)

**Comment by niklasad1** [comment]
Closing this, I found a few edge-cases which needs to be fixed before a new release.

**Pull request merged**